src/pyvmcon/vmcon.py

```diff
@@ -402,18 +395,16 @@ def _derivative_lagrangian(
     result: Result,
     lamda_equality: np.ndarray,
     lamda_inequality: np.ndarray,
-):
-    c_equality_prime = sum(
-        [lamda * dc for lamda, dc in zip(lamda_equality, result.deq)]
-    )
-    c_inequality_prime = sum(
-        [lamda * dc for lamda, dc in zip(lamda_inequality, result.die)]
-    )
+) -> np.ndarray:
+    ind_eq = min(lamda_equality.shape[0], result.deq.shape[0])
+    ind_ieq = min(lamda_inequality.shape[0], result.die.shape[0])
+    c_equality_prime = (lamda_equality[:ind_eq] * result.deq[:ind_eq]).sum(axis=0)
+    c_inequality_prime = (lamda_inequality[:ind_ieq] * result.die[:ind_ieq]).sum(axis=0)
```

Comment:
The tests fail with this for me I can't immediately see why:
![image](https://github.com/ukaea/PyVMCON/assets/81617086/97803793-1de5-4a7b-ab63-4484976947e0)

Reply:
scatch that got it working